apps/hermes/server/src/api/rest.rs

```diff
@@ -94,25 +94,45 @@ impl IntoResponse for RestError {
     }
 }
 
-/// Verify that the price ids exist in the aggregate state.
-pub async fn verify_price_ids_exist<S>(
+/// Validate that the passed in price_ids exist in the aggregate state. Return a Vec of valid price ids.
+/// # Returns
+/// If `remove_invalid` is true, invalid price ids are filtered out and only valid price ids are returned.
+/// If `remove_invalid` is false and any passed in IDs are invalid, an error is returned.
+pub async fn validate_price_ids<S>(
     state: &ApiState<S>,
     price_ids: &[PriceIdentifier],
-) -> Result<(), RestError>
+    remove_invalid: bool,
+) -> Result<Vec<PriceIdentifier>, RestError>
 where
     S: Aggregates,
 {
     let state = &*state.state;
-    let all_ids = Aggregates::get_price_feed_ids(state).await;
-    let missing_ids = price_ids
+    let available_ids = Aggregates::get_price_feed_ids(state).await;
+
+    // Find any price IDs that don't exist in the valid set
+    let not_found_ids: Vec<PriceIdentifier> = price_ids
         .iter()
-        .filter(|id| !all_ids.contains(id))
-        .cloned()
-        .collect::<Vec<_>>();
+        .filter(|id| !available_ids.contains(id))
+        .copied()
```

Comment:
what was the reason behind replacing `cloned` with `copied`? Was it just for the performance?

Reply:
I think being more explicit with copied is actually nice.

Reply:
Yeah, when i looked into `cloned` vs `copied`, copied looked more performant so just went with that